src/Npgsql/TypeHandlers/MappedEnumHandler.cs

```diff
@@ -23,35 +24,26 @@ class MappedEnumHandler<T> : NpgsqlTypeHandler<T>, IMappedEnumHandler where T :
     {
         readonly INpgsqlNameTranslator _nameTranslator;
         readonly NpgsqlConnection _conn;
-        [CanBeNull]
-        UnmappedEnumHandler _wrappedHandler;
+        readonly UnmappedEnumHandler _wrappedHandler;
 
         public Type EnumType => typeof(T);
 
-        internal MappedEnumHandler(INpgsqlNameTranslator nameTranslator, NpgsqlConnection conn)
+        internal MappedEnumHandler(INpgsqlNameTranslator nameTranslator, PostgresType pgType, NpgsqlConnection conn)
         {
+            PostgresType = pgType;
             _nameTranslator = nameTranslator;
             _conn = conn;
-        }
-
-        void WrapHandler()
-        {
             _wrappedHandler = (UnmappedEnumHandler)new UnmappedEnumTypeHandlerFactory(_nameTranslator).Create(PostgresType, _conn);
+            Debug.Assert(_wrappedHandler != null);
```

Comment:
Please remove, there's no way this assert can ever fail one line after setting _wrappedHandler